File: backend-spring/petties/src/main/java/com/petties/petties/dto/booking/EstimatedCompletionResponse.java
package com.petties.petties.dto.booking;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * Response DTO for estimated completion time
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EstimatedCompletionResponse {

    /** Start time of the booking */
    private LocalDateTime startTime;

    /** Estimated end time */
    private LocalDateTime estimatedEndTime;

    /** Total duration in minutes */
    private Integer totalDurationMinutes;

    /** Total number of slots required */
    private Integer totalSlotsRequired;

    /** Breakdown by pet */
    private List<PetDuration> pets;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class PetDuration {
        private String petId;
        private Double petWeight;
        private Integer totalDurationMinutes;
        private List<ServiceDuration> services;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ServiceDuration {
        private String serviceId;
        private String serviceName;
        private Integer durationMinutes;
        private Integer slotsRequired;
        private LocalDateTime estimatedStartTime;
        private LocalDateTime estimatedEndTime;
    }
}

File: petties_mobile/lib/data/models/estimated_completion_response.dart
/// Response from POST /bookings/public/estimated-completion
class EstimatedCompletionResponse {
  final String startTime;
  final String estimatedEndTime;
  final int totalDurationMinutes;
  final int totalSlotsRequired;

  const EstimatedCompletionResponse({
    required this.startTime,
    required this.estimatedEndTime,
    required this.totalDurationMinutes,
    required this.totalSlotsRequired,
  });

  factory EstimatedCompletionResponse.fromJson(Map<String, dynamic> json) {
    return EstimatedCompletionResponse(
      startTime: json['startTime'] as String? ?? '',
      estimatedEndTime: (json['estimatedEndTime'] ?? json['estimated_end_time']) as String? ?? '',
      totalDurationMinutes: (json['totalDurationMinutes'] as num?)?.toInt() ?? 0,
      totalSlotsRequired: (json['totalSlotsRequired'] as num?)?.toInt() ?? 0,
    );
  }
}
